### ISSUE TITLE
[postgres] Support column definition for WITH queries

### ISSUE THREAD

**Issue by ancoron** [sqlfluff/sqlfluff#345]
Please support column name definitions in queries using CTE's (common table expressions):
* https://www.postgresql.org/docs/current/sql-select.html#SQL-WITH

The following simple query fails to validate:
```sql
WITH tab_data (c_value) AS (SELECT 1)
SELECT c_value FROM tab_data;
```

...with:
```
== [db/test.sql] FAIL
L:   1 | P:   1 |  PRS | Found unparsable section: 'WITH tab_data (c_value) AS
                       | (SELECT 1)\nSE...'
```

**Comment by alanmcruickshank** [comment]
@ancoron - the part that it won't be liking is the `(c_value)` after `tab_data`. It's not expecting to find a bracketed expression there.  In the link your provided I can't see where that syntax is documented. The only additional syntax I could see is the `RECURSIVE` options which I think we should address separately.

The query without that parses as expected:
```
WITH tab_data AS (SELECT 1)
SELECT c_value FROM tab_data;
```

**Comment by ancoron** [comment]
In the section I linked, the second paragraph:
> A name (without schema qualification) must be specified for each WITH query. **Optionally, a list of column names can be specified**; if this is omitted, the column names are inferred from the subquery.

**Comment by barrywhart** [comment]
Apparently, optional column lists in a CTE are part of the ANSI-99 specification, so they are not specific to Postgres.

* Wikipedia article which I believe is documenting the ANSI-99 standard: https://en.wikipedia.org/wiki/Hierarchical_and_recursive_queries_in_SQL#Common_table_expression
* SQL Server docs: https://docs.microsoft.com/en-us/sql/t-sql/queries/with-common-table-expression-transact-sql?view=sql-server-ver15
* MySQL docs: https://www.mysqltutorial.org/mysql-cte/

But apparently it is not universal, either, as I see no mention of it in the BigQuery [documentation](https://cloud.google.com/bigquery/docs/reference/standard-sql/query-syntax).

**Comment by alanmcruickshank** [comment]
This was addressed in #684. I'm going to close this issue.
